packages/eslint-plugin/src/rules/type-annotation-spacing.ts

```diff
@@ -32,6 +47,107 @@ const definition = {
   additionalProperties: false,
 };
 
+function createRules(options?: Config): WhitespaceRules {
+  const globals = {
+    ...(options?.before !== undefined ? { before: options.before } : {}),
+    ...(options?.after !== undefined ? { after: options.after } : {}),
+  };
+  const override = options?.overrides ?? {};
+  const colon = {
+    ...{ before: false, after: true },
+    ...globals,
+    ...override?.colon,
+  };
+  const arrow = {
+    ...{ before: true, after: true },
+    ...globals,
+    ...override?.arrow,
+  };
+
+  return {
+    colon: colon,
+    arrow: arrow,
+    variable: { ...colon, ...override?.variable },
+    property: { ...colon, ...override?.property },
+    parameter: { ...colon, ...override?.parameter },
+    returnType: { ...colon, ...override?.returnType },
+  };
+}
+
+function getNodeType(node?: TSESTree.Node): string {
+  return node?.type ?? '';
+}
+
+function typeIsArrowFunction(type: string): boolean {
+  return type === AST_NODE_TYPES.TSFunctionType;
+}
```

Comment:
You should move this to `utils/astUtils.ts`, and instead make it a type guard on the node directly

```ts
function isTSFunctionType(
  node: TSESTree.Node | undefined,
): node is TSESTree.TSFunctionType {
  return node?.type === AST_NODE_TYPES.TSFunctionType;
}
```

Reply:
done